src/main/java/de/siegmar/logbackgelf/InetUtil.java

```diff
@@ -20,13 +20,41 @@
 package de.siegmar.logbackgelf;
 
 import java.net.InetAddress;
+import java.net.NetworkInterface;
 import java.net.UnknownHostException;
+import java.util.Enumeration;
+import java.util.LinkedHashSet;
 
 final class InetUtil {
 
     private InetUtil() {
     }
 
+    /**
+     * Set this system property to a simple hostname to use a fixed host name.
+     */
+    public static final String PROPERTY_LOGSTASH_GELF_HOSTNAME = "logstash-gelf.hostname";
+
+    static String sysEnv (String envVarName){
+        try {
+            String value = System.getProperty(envVarName);
+            if (value != null && !value.isBlank()){
+                return value;
+            }
+        } catch (Throwable ignore){
+        }
+        try {
+            return System.getenv(envVarName);
+        } catch (Exception ignore){
+        }
+        return null;
+    }
+
+    static String trim (String str){
+        return str == null ? ""
+            : str.trim().strip();
+    }
+
     /**
      * Retrieves the local host's name. Preferably the fully qualified one.
```

Comment:
Javadoc doens't seem to match the implementation now. Tests are missing.

Reply:
This is copt paste from my working tested code. There is additional test, you named it "unrelated" :-)

Your original compact version is broken in case of Docker installed: it shows not a real hostname, but something like internal.docker.host

The new version also supports explicit hostname specification (through system property)